lib/deltachat.ts

```diff
@@ -574,7 +576,7 @@ export class DeltaChat extends EventEmitter {
   // https://c.delta.chat/classdc__context__t.html#ae49176cbc26d4d40d52de4f5301d1fa7
   joinSecurejoin (qrCode:string) {
     debug(`joinSecurejoin ${qrCode}`)
-    return binding.dcn_join_securejoin(this.dcn_context, qrCode)
+    return binding.dcn_join_secure(this.dcn_context, qrCode)
```

Comment:
You need to pass the callback here as the third argument ;) Otherwise the NAPI_ASYNC_QUEUE_WORK macro can't get the second element of the argv array.

Reply:
Function should look like this:
```
  joinSecurejoin (qrCode:string: callback: (chatId: Number) => void) {
    debug(`joinSecurejoin ${qrCode}`)
    return binding.dcn_join_secure(this.dcn_context, qrCode)
}
```

Reply:
Added it but still get the same error: 
napi_create_reference(env, callback, 1, &carrier->callback_ref) failed!

Maybe the NAPI_ASYNC_QUEUE_WORK macro does not work for this use case?

Reply:
Are you sure that you rebuilt the typescript code? Added a unit test which passes (and the callback is getting called)